reconcile/test/test_terraform_vpc_peerings.py

```diff
@@ -10,57 +13,309 @@
 
 
 class MockOCM:
-    @staticmethod
-    def get_aws_infrastructure_access_terraform_assume_role(cluster,
+
+    def __init__(self) -> None:
+        self.assumes: dict[str, str] = {}
+
+    def register(self, cluster: str, tf_account_id: str, tf_user: str,
+                 assume_role: Optional[str]) -> 'MockOCM':
+        if not assume_role:
+            assume_role = f"arn::::{cluster}"
+        if not assume_role.startswith("arn:"):
+            assume_role = f"arn::::{assume_role}"
+        self.assumes[f"{cluster}/{tf_account_id}/{tf_user}"] = assume_role
+        return self
+
+    def get_aws_infrastructure_access_terraform_assume_role(self, cluster,
                                                             tf_account_id,
                                                             tf_user):
-        return f"{cluster}/{tf_account_id}/{tf_user}"
+        return self.assumes.get(f"{cluster}/{tf_account_id}/{tf_user}")
 
 
-class TestAWSAccountFromInfrastructureAccess(testslide.TestCase):
-    def setUp(self):
-        self.cluster = {
-            'name': 'cluster',
-            'spec': {
-                'region': 'region'
-            },
-            'network': {
-                'vpc': 'vpc'
-            },
-            'awsInfrastructureAccess': [
-                {
-                    'awsGroup': {
-                        'account': {
-                            'name': 'account',
-                            'uid': 'uid',
-                            'terraformUsername': 'terraform',
-                            'automationToken': 'token'
-                        }
+class MockAWSAPI:
```

Comment:
While I agree this helps remove testslide, it risks diverging from the stubbed code. If the API changes this stub will stay behind and these tests won't resemble reality anymore.